File: ruby/lib/pagou/format.rb
# frozen_string_literal: true

require "json"

module Pagou
  # Formats an integer amount in the smallest currency unit as a display string.
  def self.format_amount(cents, currency = "BRL")
    symbol = { "BRL" => "R$", "USD" => "$" }.fetch(currency, "#{currency} ")
    formatted = format("%.2f", cents.to_f / 100).reverse.gsub(/(\d{3})(?=\d)/, '\1.').reverse
    "#{symbol}#{formatted}"
  end

  # A short, unique idempotency key for a given operation and reference.
  def self.idempotency_key(operation, reference)
    "#{operation}_#{reference}"
  end

  # Prints a labelled JSON block for readable script output.
  def self.print_result(label, value)
    puts "\n#{label}:"
    puts JSON.pretty_generate(value)
  end
end

File: python/pagou/format.py
import json
from decimal import Decimal
from typing import Any

_CURRENCY_SYMBOLS = {"BRL": "R$", "USD": "$", "MXN": "$", "ARS": "$", "COP": "$", "CLP": "$"}


def format_amount(cents: int, currency: str = "BRL") -> str:
    """Formats an integer amount in the smallest currency unit as a display string."""
    symbol = _CURRENCY_SYMBOLS.get(currency, f"{currency} ")
    value = (Decimal(cents) / Decimal(100)).quantize(Decimal("0.01"))
    return f"{symbol}{value:,.2f}"


def idempotency_key(operation: str, reference: str) -> str:
    """A short, unique idempotency key for a given operation and reference."""
    return f"{operation}_{reference}"


def print_result(label: str, value: Any) -> None:
    """Prints a labelled JSON block for readable script output."""
    print(f"\n{label}:")
    print(json.dumps(value, indent=2, default=str))
